agentos/component.py

```diff
@@ -223,3 +141,111 @@ def _parse_components(cls, components_spec: Dict, repos: Dict) -> Dict:
                 component.add_dependency(dependency, alias=alias)
 
         return components
+
+    def run(self, fn_name: str, params: ParameterSet = None):
+        params = params if params else ParameterSet()
+        with self._track_run(fn_name, params):
+            return self._untracked_run(fn_name, params)
+
+    def _untracked_run(
```

Comment:
As we discussed in-person, can we get rid of `_untracked_run()` and instead have `run()` take a `tracked=False` flag?